tutorials/kotlin-springboot-container-engine.md

```diff
@@ -0,0 +1,383 @@
+It then copies the sources from the local file system (which should be your project folder) to the container. It sets the
+working directory to the `work` folder created and then runs the Gradle wrapper command (`gradlew`) to build the project. This wrapper
+automatically downloads Gradle and then builds the project. Once built, it then copies the output of the `jar` to the `work` folder
+ready for execution when invoking `docker run`.
+
+### Test the Dockerfile
+
+Build the image locally and test running your application from the image:
+
+    docker build --no-cache -t demo .
+    docker run -it --rm -p 8080:8080 demo
+
+The period at the end of the `docker build` command is required. It denotes the
+root directory of the application you are building.
+
+Visit [http://localhost:8080/message](http://localhost:8080/message) to see the application
+respond running locally from your Docker image.
+
+## Deploying your application
+
+Now you're ready to deploy your application to Kubernetes Engine!
+
+### Build the production image
+
+To deploy the app, you will use the
+[Google Cloud Container Build](https://cloud.google.com/container-builder/)
+service to build your Docker image in the cloud and store the resulting Docker
+image in your project in the
+[Google Cloud Container Registry](https://cloud.google.com/container-registry/).
+
+Execute the following command to run the build:
+
+    gcloud container builds submit --tag=gcr.io/${PROJECT_ID}/demo:v1 .
+
+Replace `${PROJECT_ID}` with the ID of your Google Cloud Platform project.
+The period at the end is required.
+
+After the build finishes, the image `gcr.io/${PROJECT_ID}/demo:v1` will be
+available. You can list the images you have built in your project using:
+
+    gcloud container images list
+
+You can even push and pull the image directly from your registry. See the
+[Container Registry how-to guides](https://cloud.google.com/container-registry/docs/pushing-and-pulling)
+for more details.
+
+### Create a cluster
+
+Kubernetes Engine lets you create Kubernetes clusters to host your application.
+These are clusters of VMs in the cloud, managed by a Kubernetes server.
+
+1.  Choose a cluster name. For the rest of these instructions, I'll assume that
+    name is "demo-cluster".
+
+1.  Create the cluster.
+
+        gcloud container clusters create demo-cluster --num-nodes=2
+
+    This command creates a cluster of two machines. You can choose a different
+    size, but two is a good starting point.
+
+    It might take several minutes for the cluster to be created. You can check
+    the cloud console at http://cloud.google.com/console, under the Kubernetes
+    Engine section, to see that your cluster is running. You will also be able
+    to see the individual running VMs under the Compute Engine section. Note
+    that once the cluster is running, you will be charged for the VM usage.
+
+1.  Configure the gcloud command-line tool to use your cluster by default, so
```

Comment:
Configure the `gcloud` command-line tool to use your cluster by default, so you don't have to specify it every time for the remaining `gcloud` commands.